### PR TITLE
Define the "dhcp_relay" model for the Dell Enterprise SONiC collection

### PR DESCRIPTION
**Pull request by ArunSaravananBalachandran**
Initial posting for the "dhcp_relay" model definition.

### CODE REVIEW

**Review by kerry-meyer — COMMENTED** [review]
This change set looks good. I see no problems in the model ".yml" file.

The only modification I would suggest is to add a "deleted" example between the current first and second deletion examples to show how to delete dhcp_relay configuration for one of the two address families on Eth1/1.

Note: There is a failing sanity check ("ansible/third-party-check") for this pull request. Looking at the details, it looks like it may be an Ansible problem that isn't caused by anything wrong in the proposed change set. After making the suggested change and pushing it to the branch, a new sanity run will be triggered. Hopefully, the Ansible core team will have fixed the problem by then. If not, I'll check with the Ansible core team to find out what should be done to fix the problem.

File: models/enterprise_sonic/dhcp_relay/deleted_example_02.txt
Comment (kerry-meyer) [inline models/enterprise_sonic/dhcp_relay/deleted_example_02.txt:38]: The presence of this line, with nothing after it, should be sufficient for deletion of all ipv4 dhcp_relay configuration on this interface. (If the list of server addresses is explicitly deleted, as shown here, other attributes would remain present after processing the deletion. Don't worry about that case in the deletion examples, though. I am just requesting that the following two lines be deleted in this example.)

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
Thanks for specifying the additional cases.

I have two minor comments on this version, but they are both one or two line changes.

File: models/enterprise_sonic/dhcp_relay/deleted_example_02.txt
Comment (kerry-meyer) [reply models/enterprise_sonic/dhcp_relay/deleted_example_02.txt:38]: We had a meeting today to discuss the implementation details and problem that you posted via email and have done more research on existing precedents. I understand that your proposed implementation will work for doing bulk deletion of the defined servers and the way that I requested will not work for this model. For now, it is okay to keep this part of the implementation as you have proposed here. I will mark this issue as "Resolved".

*********************************************

Note:

For possible future consideration: There is at least one existing case in the current code that may provide a way to slightly simplify the user interface for bulk deletion: 

l2_interfaces: Deletion of all "trunk" vlans [or the "access" vlan] by specifying a key with no following value.

https://github.com/ansible-collections/dellemc.enterprise_sonic/blob/main/plugins/module_utils/network/sonic/config/l2_interfaces/l2_interfaces.py  (lines 346-361):

```
                    # check for access or trunk is mentioned without value
                    else:
                        # access mentioned wothout value
                        if ('access' in keys) and conf.get('access', None) is None:
                            # get the existing values and delete it
                            if matched.get('access'):
                                conf['access'] = matched.get('access').copy()
                                request = self.get_access_delete_switchport_request(conf, matched)
                                if request:
                                    requests.append(request)
                        # trunk mentioned wothout value
                        if ('trunk' in keys) and conf.get('trunk', None) is None:
                            # get the existing values and delete it
                            if matched.get('trunk'):
                                conf['trunk'] = matched.get('trunk').copy()
                                requests.extend(self.get_trunk_delete_switchport_request(conf, matched))
```

We would like to see if the code shown above works as intended and, if so, why the behavior differs from what is being seen with the current DHCP relay model. But I will take this as an action item and will look into as time permits. We don't want to delay the DHCP Relay implementation while this issue is being investigates, so please proceed with implementation of this model and resource module as you have currently proposed it.

File: models/enterprise_sonic/dhcp_relay/deleted_example_02.txt
Comment (ArunSaravananBalachandran) [reply models/enterprise_sonic/dhcp_relay/deleted_example_02.txt]: As suggested, have added 'IPv4' in the name.

**Review by kerry-meyer — APPROVED** [review]
Thanks for providing this new model and making the subsequent changes.

The current proposed change set looks good.

I am approving the PR, but will still need to resolve the sanity failure before we can get the PR merged. I'll keep you posted on the status for that problem.

File: models/enterprise_sonic/dhcp_relay/deleted_example_02.txt
Comment (kerry-meyer) [reply models/enterprise_sonic/dhcp_relay/deleted_example_02.txt]: Thank you. I am marking the PR "approved".

**Comment by NilashishC** [comment]
recheck

**Pull request merged**
